components/concept/storage/src/main/java/mozilla/components/concept/storage/CreditCardsStorage.kt

```diff
@@ -0,0 +1,202 @@
+     * @param guid Unique identifier for the desired address.
+     * @return [Address] record.
+     */
+    suspend fun getAddress(guid: String): Address
+
+    /**
+     * Retrieves a list of all the addresses.
+     *
+     * @return A list of all [Address].
+     */
+    suspend fun getAllAddresses(): List<Address>
+
+    /**
+     * Updates the fields in the provided address.
+     *
+     * @param address An [Address] record to update.
+     */
+    suspend fun updateAddress(address: Address)
+
+    /**
+     * Delete the address with the given [guid].
+     *
+     * @return True if the deletion did anything, false otherwise.
+     */
+    suspend fun deleteAddress(guid: String): Boolean
+
+    /**
+     * Marks the address with the given [guid] as `in-use`.
+     *
+     * @param guid Unique identifier for the desired address.
+     */
+    suspend fun touchAddress(guid: String)
+}
+
+/**
+ * Information about a credit card.
+ *
+ * @property guid The unique identifier for this credit card.
+ * @property ccName The credit card billing name.
+ * @property ccNumber The credit card number.
+ * @property ccExpMonth The credit card expiry month.
+ * @property ccExpYear The credit card expiry year.
+ * @property ccType The credit card network ID.
+ */
+data class CreditCard(
+    val guid: String,
+    val ccName: String,
+    val ccNumber: String,
+    val ccExpMonth: Long,
+    val ccExpYear: Long,
+    val ccType: String
+)
+
+/**
+ * Information about a new credit card.
+ *
+ * @property ccName The credit card billing name.
+ * @property ccNumber The credit card number.
+ * @property ccExpMonth The credit card expiry month.
+ * @property ccExpYear The credit card expiry year.
+ * @property ccType The credit card network ID.
+ */
+data class NewCreditCardFields(
+    val ccName: String,
+    val ccNumber: String,
+    val ccExpMonth: Long,
+    val ccExpYear: Long,
+    val ccType: String
+)
+
+/**
+ * Information about a address.
+ *
+ * @property guid The unique identifier for this address.
+ * @property givenName First name.
+ * @property additionalName Middle name.
+ * @property familyName Last name.
+ * @property organization Organization.
+ * @property streetAddress Street address.
+ * @property addressLevel3 Sublocality (Suburb) name type.
+ * @property addressLevel2 Locality (City/Town) name type.
+ * @property addressLevel1 Province/State name type.
+ * @property postalCode Postal code.
+ * @property country Country.
+ * @property tel Telephone number.
+ * @property email E-mail address.
+ */
+data class Address(
+    val guid: String,
+    val givenName: String,
+    val additionalName: String,
+    val familyName: String,
+    val organization: String,
+    val streetAddress: String,
+    val addressLevel3: String,
+    val addressLevel2: String,
+    val addressLevel1: String,
+    val postalCode: String,
+    val country: String,
+    val tel: String,
+    val email: String
+)
+
+/**
+ * Information about a new address.
```

Comment:
Can you expand this a bit to explain why we have two nearly identical classes? I think with an expanded comment and renaming of this class (e.g. UpdatableAddressFields), it should be pretty clear what's going on.